### PR TITLE
Enable wire hover detection in global hover resolution

### PR DESCRIPTION
**Pull request by yashveer-1**
Fixes #

#### Describe the changes you have made in this PR -
This PR adds wire hover detection to the simulator’s global hover resolution.

Previously, `simulationArea.hover` was only set for nodes and circuit elements.  
Wires were never considered during hover detection, so the cursor and interaction
logic could not correctly reflect when the user was hovering a wire.

This change updates the engine’s update cycle to:
- Check whether the mouse is over any wire
- Set `simulationArea.hover` to that wire when appropriate

This enables consistent cursor feedback and provides the foundation
for visual wire hover effects in future improvements.

### Screenshots of the UI changes (If any) -
WhatsApp Image 2026-01-11 at 18 34 36 (1)
WhatsApp Image 2026-01-11 at 18 34 36

---

## Code Understanding and AI Usage

- [x] Yes, I used AI assistance  
- [ ] No, I wrote all the code myself  

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code  
- [x] I can explain the purpose and logic of each function/component I added  
- [x] I have tested edge cases and understand how the code handles them  
- [x] I have modified the AI output to follow this project's coding standards and conventions  

**Explain your implementation approach:**

**Problem:**  
Wires were not part of the global hover resolution system. The simulator only
checked nodes and circuit elements, so hovering over a wire did not register as
a hover target even though the cursor was expected to change.

**Approach:**  
Instead of adding hover logic inside `Wire.draw` or `Wire.update`, I added wire
hover detection at the engine level where `simulationArea.hover` is resolved.
This keeps all hover ownership logic in one place and avoids conflicts with
existing element and node hover behavior.

**Implementation:**  
During each update cycle (when the mouse is not down), the engine iterates over
all wires and checks whether the mouse position lies on a wire using  
`wire.checkWithin(x, y)`. If a match is found, `simulationArea.hover` is set to
that wire.

This integrates wires into the same hover pipeline used by nodes and elements,
making cursor behavior and future visual hover effects consistent.

I chose this approach because it matches how the rest of the simulator resolves
hover state and avoids duplicating hover logic inside individual draw methods.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue  
- [x] I have performed a self-review of my code  
- [x] **I can explain the purpose of every function, class, and logic block I added**  
- [x] I understand why my changes work and have tested them thoroughly  
- [x] I have considered potential edge cases and how my code handles them  
- [ ] If it is a core feature, I have added thorough tests  
- [x] My code follows the project's style guidelines and conventions  

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed wire hover detection to properly respond when hovering over wires.
  * Improved drag rectangle selection to correctly include all objects within the selected area.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: v0/src/simulator/src/engine.js [hunk 1]
```diff
@@ -529,6 +529,18 @@ export function update(scope = globalScope, updateEverything = false) {
...
+    //Wire hover resolution
+if (!simulationArea.mouseDown) {
+    for (let i = 0; i < scope.wires.length; i++) {
+        const w = scope.wires[i]
+        if (w.checkWithin(simulationArea.mouseX, simulationArea.mouseY)) {
+            simulationArea.hover = w
+            break
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (yashveer-1) [reply v0/src/simulator/src/engine.js]: Thanks for the review!
I’ve updated the wire hover logic to guard against overriding existing node/element hover (simulationArea.hover === undefined), added proper indentation, and iterated wires in reverse order so topmost wires take priority.

This keeps hover resolution consistent with the existing pipeline and avoids conflicts with selection and pane logic.
